web/src/features/time/TimeController.tsx

```diff
@@ -102,27 +102,38 @@ export default function TimeController({ className }: { className?: string }) {
 
   const onToggleGroupClick = useCallback(
     (timeRange: TimeRange) => {
-      // Set time slider to latest value before switching aggregate to avoid flickering
-      setSelectedDatetime({
-        datetime: selectedDatetime.datetime,
-        index: numberOfEntries,
-      });
-      setTimeRange(timeRange);
-      trackEvent(TrackEvent.TIME_AGGREGATE_BUTTON_CLICKED, { timeRange });
+      if (datetimes !== undefined) {
+        const lastDatetime = datetimes.at(-1);
+        if (lastDatetime) {
+          // Set time slider to latest value before switching aggregate to avoid flickering
+          setSelectedDatetime({
+            datetime: lastDatetime,
+            index: numberOfEntries,
+          });
+          setTimeRange(timeRange);
+          trackEvent(TrackEvent.TIME_AGGREGATE_BUTTON_CLICKED, { timeRange });
+        }
+      }
     },
-    [setSelectedDatetime, selectedDatetime.datetime, numberOfEntries, setTimeRange]
+    [setSelectedDatetime, datetimes, numberOfEntries, setTimeRange]
+  );
+
+  // This is required to avoid rerending the toggle needlessly
+  const timeRangeToggle = useMemo(
+    () => (
+      <TimeRangeToggle
+        timeRange={selectedTimeRange || TimeRange.H24}
+        onToggleGroupClick={onToggleGroupClick}
+      />
+    ),
+    [selectedTimeRange, onToggleGroupClick]
```

Comment:
Same here as with the other comment, I think it would be best to memo the component itself and not use a useMemo hook with it.

Reply:
Indeed this works. Will change